### DISCUSSION TITLE
"One Click Transcribe" tool

### DISCUSSION THREAD

**Discussion by Purfview** [Purfview/whisper-standalone-win#337] [Show and tell]
```


     8""""                                    8   8  8                                       8   8 8   8 8
     8     eeeee eeeee eeeee eeee eeeee       8   8  8 e   e e  eeeee eeeee eeee eeeee        8 8   8 8  8
     8eeee 8   8 8   "   8   8    8   8       8e  8  8 8   8 8  8   " 8   8 8    8   8        eee   eee  8e
     88    8eee8 8eeee   8e  8eee 8eee8e eeee 88  8  8 8eee8 8e 8eeee 8eee8 8eee 8eee8e eeee 88  8 88  8 88
     88    88  8    88   88  88   88   8      88  8  8 88  8 88    88 88    88   88   8      88  8 88  8 88
     88    88  8 8ee88   88  88ee 88   8      88ee8ee8 88  8 88 8ee88 88    88ee 88   8      88  8 88  8 88eee


Nothing to do. Read usage instruction at this link:
https://github.com/Purfview/whisper-standalone-win/discussions/337

Press any key to continue . . .
```

Some people don't understand power of CLI, they want useless GUI. So I added `One Click Transcribe.bat` to a pack.

How to utilize it:

1) Press the right mouse button on the "One Click Transcribe.bat" file.
2) Then in the context menu select `Send To` > `Desktop (create shortcut)` to create a shortcut on your Desktop.

Now you can drag and drop file or files or folder onto this shortcut to do automatic one click transcriptions.

 "One Click Transcribe.bat" is just a text file, you can edit it with Notepad and add there the settings you prefer.

## Note:
When task is to transcribe then language should be set to activate custom prompt presets, so if you leave it on auto then those presets won't be activated and transcription result may have worse punctuation ect.

# Linux steps
1) Open terminal in the same folder where executable is.
2) Copy/paste the command below and press Enter, a shortcut should appear on a Desktop:

```
chmod +x faster-whisper-xxl

cat <<EOF > ~/Desktop/One_Click_Transcribe.desktop
[Desktop Entry]
Version=1.0
Type=Application
Name=One Click Transcribe
Comment=Process audio files with One_Click_Transcribe.sh
Exec=$(pwd)/One_Click_Transcribe.sh %F
Icon=audio-x-generic
Terminal=true
Path=$(pwd)
MimeType=audio/x-wav;audio/mpeg;audio/x-flac;audio/x-m4a;
Categories=Audio;Utility;
EOF

chmod +x ~/Desktop/One_Click_Transcribe.desktop
```

**Comment by vimbackground** [comment]
thx, where is the file?

**Reply by Purfview** [reply]
In the  r194.2 release.

**Comment by defaultsecurity** [comment]
Awesome. You have read my mind. I was just about to write such a script. Impeccable timing and the script is excellent. Thank you very much.

**Comment by dragokas** [comment]
How can I send pull request for this script? It has parsing bug.

**Reply by Purfview** [reply]
Post here about it and the fix.

**Reply by dragokas** [reply]
You're trying to make each argument to be safe by enquoting, but this construction is unsafe itself: for %%F in (%*) 

Fixing:
 - Crash on filenames having "(", ")" characters.
 - Unable to process filenames with "!" characters.
 - Possible current path issues.

<details>
  <summary>Code:</summary>
  
```
@echo off
color 0A
echo.
echo.
echo      8""""                                    8   8  8                                       8   8 8   8 8     
echo      8     eeeee eeeee eeeee eeee eeeee       8   8  8 e   e e  eeeee eeeee eeee eeeee        8 8   8 8  8     
echo      8eeee 8   8 8   "   8   8    8   8       8e  8  8 8   8 8  8   " 8   8 8    8   8        eee   eee  8e    
echo      88    8eee8 8eeee   8e  8eee 8eee8e eeee 88  8  8 8eee8 8e 8eeee 8eee8 8eee 8eee8e eeee 88  8 88  8 88    
echo      88    88  8    88   88  88   88   8      88  8  8 88  8 88    88 88    88   88   8      88  8 88  8 88    
echo      88    88  8 8ee88   88  88ee 88   8      88ee8ee8 88  8 88 8ee88 88    88ee 88   8      88  8 88  8 88eee 
echo.
echo.                                                                                                            

:: Localize variables, enable extended syntax
setlocal enableextensions

:: Check if files were dropped onto the batch file
if "%~1"=="" (
    echo Nothing to do. Read usage instruction at this link:
    echo https://github.com/Purfview/whisper-standalone-win/discussions/337
    echo.
    pause
    color
    exit /b
)

:: Tool's path
set "dp=%~dp0"

:: Initialize file list
set "file_list="

:: Enquote all file paths
:while
for %%F in ("%~1") do set "file_list=%file_list% "%%~F""
shift
if "%~1" neq "" goto while

:: Start processing
"%dp%faster-whisper-xxl.exe" %file_list% -pp -o source --batch_recursive --check_files --standard -f json srt -m medium


pause
color
exit /b
```
  
</details>

**Reply by Purfview** [reply]
Thx. I'll add fixes in the next version.

**Comment by Soeroah** [comment]
Hi,

Thanks for this option. I tried to get it working with just my old batch file but it didn't want to play nice.

Could someone please help me tweak the one click batch file to output similar to what this older one did, plus speaker diarization?

> C:\Users\David\Downloads\Faster-Whisper-XXL_r192.3.4_windows\Faster-Whisper-XXL\faster-whisper-xxl.exe "C:\Users\David\Documents\SoundRecordMp3\SoundRecordMp3\bin\Debug\Output" --language=en --model=medium --batch_recursive --beam_size=1 --output_format text --output_dir "C:\Users\David\Documents\SoundRecordMp3\SoundRecordMp3\bin\Debug\Output"

If I'm dragging the folder onto the batch file, I imagine I don't need the first line (where the .exe file resides) or the folder where the files I want are. The options, though, I'm a little confused where to put them in the batch. My gut is to replace

> faster-whisper-xxl.exe %file_list% -pp -o source --batch_recursive --check_files --standard -f json srt -m medium

at the bottom of the text file with everything from --language on, adding the --diarize filter, but I'm not sure if the other stuff has changed between 192.3.4 and 245.2.0.0. 

The core things I want are to output as a .text file in that particular output folder, and to have diarization enabled for at least english. Thank you

**Reply by Purfview** [reply]
>If I'm dragging the folder onto the batch file,

Don't drop onto the bat file directly, drop onto its shortcut.

>Could someone please help me tweak the one click batch file to output similar to what this older one did, plus speaker diarization?

Try this:
```
faster-whisper-xxl.exe %file_list% -o source -br -bs 1 -f text -m medium -l en --diarize pyannote_v3.0
```

**Reply by Soeroah** [reply]
Right, the shortcut, that's what I meant

Thank you, I'll give that a try once it finishes downloading and installing the first time models

**Reply by Purfview** [reply]
>downloading and installing the first time models

You don't need to re-download them, just copy the old ones.

Btw, I didn't included these commands: `-pp  --check_files --standard`, and you may want to use `--sentence`

>--beam_size=1 

I guess you use this option [short version of it: `-bs 1` ] because you run it on CPU, try `--batched` mode to make it faster.
Note: `--batched`  is not related to batch file processing.

**Reply by Soeroah** [reply]
So the whole command should be

> faster-whisper-xxl.exe %file_list% -o source -br -bs 1 -f text -m medium -l en --diarize pyannote_v3.0 -pp  --check_files --standard --sentence --beam_size=1 --batched

Is that right?

And ah, I didn't know about copying the old models, but I don't know where those are installed or where to put them either and I'm up to 70% progress now

**Reply by Purfview** [reply]
>Is that right?

If you use `--standard` then `--sentence` is redundant.
You are using `--beam_size=1` twice as `-bs 1`  is the same thing.

>I didn't know about copying the old models, but I don't know where those are installed or where to put them either

Models are at a "exe" file, in "_models" folder.

**Comment by lkeels** [comment]
So how is this better than a GUI where you could just drop a file and be done? This is like Egyptian hieroglyphics.

**Reply by Purfview** [reply]
>So how is this better than a GUI where you could just drop a file and be done?

GUI would be "Three-Clicks", so you can do the math on how much better "One-Click" is. 😁

>This is like Egyptian hieroglyphics.

This is just basic computer usage. What difficulties do you have with it?

**Reply by Purfview** [reply]
Maybe you'll like "Two-Clicks" context menu tool -> https://github.com/Purfview/whisper-standalone-win/discussions/539

**Comment by ManekiNeko-89** [comment]
Hello. Sorry at bother you, but how do we get the transcript translated using this method/tool, please?

**Reply by Purfview** [reply]
Add `--task translate` to the command

**Reply by ManekiNeko-89** [reply]
> Add `--task translate` to the command

You mean if I open the .bat file in Notepad++? 
<img width="1178" height="815" alt="Capture d&#39;écran 2025-12-27 213514" src="https://github.com/user-attachments/assets/6ae99cff-8465-480c-b44c-b0dc11c6ecec" />

Because I can't typed anything if I open the bat file or drag a video onto it.

**Reply by Purfview** [reply]
>You mean if I open the .bat file in Notepad++?

Yes.

**Reply by ManekiNeko-89** [reply]
> > You mean if I open the .bat file in Notepad++?
> 
> Yes.

Alright, thank you so much for your kindness in taking the time to answer my questions.
